src/base.jl

```diff
@@ -31,13 +33,27 @@ const Components = Dict{DataType, Dict{String, <:Component}}
     # Keyword arguments
 
     * `runchecks`::Bool : run available checks on input fields
+    * `configpath`::String : specify path to validation config file
     DOCTODO: any other keyword arguments? genmap_file, REGEX_FILE
 """
 struct System <: PowerSystemType
     components::Components
     forecasts::SystemForecasts
     basepower::Float64             # [MVA]
     internal::PowerSystemInternal
+    validation_descriptor::Vector
+
+    function System(components, forecasts, basepower, internal; kwargs...)
+        configpath = get(kwargs, :configpath, VALID_CONFIG_FILE)
+        runchecks = get(kwargs, :runchecks, true)
+        if runchecks
+            validation_descriptor = read_validation_descriptor(configpath)
+            sys = new(components, forecasts, basepower, internal, validation_descriptor)
+        else
+            sys = new(components, forecasts, basepower, internal, Vector())
+        end
```

Comment:
An explicit return statement would be good here.

Reply:
Also, if `kwargs` aren't being propagated further, can we not explicitly right them out?

```
function System(components, forecasts, basepower, internal; runchecks=true, configpath=VALID_CONFIG_FILE, kwargs...)
...
end
```